rosbridge_library/test/capabilities/test_call_service.py

```diff
@@ -3,108 +3,168 @@
 import unittest
 from json import dumps, loads
 
-import rospy
-import rostest
+import rclpy
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
 from rosbridge_library.capabilities.call_service import CallService
-from rosbridge_library.protocol import (
+from rosbridge_library.internal.exceptions import (
     InvalidArgumentException,
     MissingArgumentException,
-    Protocol,
 )
-from roscpp.srv import GetLoggers
-from std_srvs.srv import SetBool
+from rosbridge_library.protocol import Protocol
+from std_srvs.srv import SetBool, Trigger
 
 
 class TestCallService(unittest.TestCase):
+    def trigger_cb(self, request, response):
+        """Helper callback function for a test service with no arguments."""
+        response.success = True
+        response.message = "called trigger service successfully"
+        return response
+
+    def set_bool_cb(self, request, response):
+        """Helper callback function for a test service with arguments."""
+        response.success = request.data
+        if request.data:
+            response.message = "set bool to true"
+        else:
+            response.message = "set bool to false"
+        return response
+
     def setUp(self):
-        rospy.init_node("test_call_service")
+        rclpy.init()
+        self.executor = MultiThreadedExecutor()
```

Comment:
nit: but would avoid a os call in these unit tests. Could maybe put it in all these but will limit the number of comments I make. Though stupid question for my own edification, would all these tests work with a single threaded executor? Or would everything just hang?

```suggestion
        self.executor = MultiThreadedExecutor(num_threads=2)
```

Reply:
I just checked by switching everything over to `SingleThreadedExecutor` and most of it still worked, except for one test. For that one, I used your suggestion.